### ISSUE TITLE
Steady State Simulation fails

### ISSUE THREAD

**Issue by jvanhoefer** [AMICI-dev/AMICI#1326]
Optimization in pyPESTO fails due to `Steady state computation failed.`

```
[Warning] AMICI:NaN: AMICI encountered a NaN value at index 10/29 in fxdot!
[Warning] AMICI:NaN: AMICI encountered a NaN value at index 12/122 in p!
[Warning] AMICI:NaN: AMICI encountered a NaN value at index 0/16 in w!
[Warning] AMICI:CVODES:CVode:OTHER: AMICI ERROR: in module CVODES in function CVode : The right-hand side routine failed at the first call. 
[Warning] AMICI:simulation: AMICI simulation failed:
Steady state computation failed. First run of Newton solver failed: No convergence was achieved. Simulation to steady state failed. Second run of Newton solver failed: No convergence was achieved.
Error occured in:
0          0x1114a5d03 amici::SteadystateProblem::handleSteadyStateFailure(amici::Solver const*, amici::Model*) + 531
1          0x1114a52fc amici::SteadystateProblem::findSteadyState(amici::Solver*, amici::NewtonSolver*, amici::Model*, int) + 332
2          0x1114a504e amici::SteadystateProblem::workSteadyStateProblem(amici::Solver*, amici::Model*, int) + 254
3          0x1114866e6 amici::AmiciApplication::runAmiciSimulation(amici::Solver&, amici::ExpData const*, amici::Model&, bool) + 230
4  
```
When simulating for a long time the system seemed to reach steady, though.

**AMICI version and system environment**
 - iOS
 - current `develop` branch of python AMICI
[Pom1p_PEtab.zip](https://github.com/AMICI-dev/AMICI/files/5452770/Pom1p_PEtab.zip)

**Comment by dweindl** [comment]
Did you have a look at the state trajectories -  is it safe to assume that there is a steady state for those parameters?

**Comment by jvanhoefer** [comment]
Good point, I did not. But at least it fails for all starts, that I ran so far.

**Comment by jvanhoefer** [comment]
I will have a look and update here

**Comment by FFroehlich** [comment]
> [Warning] AMICI:NaN: AMICI encountered a NaN value at index 12/122 in p!
This looks fishy

**Comment by dweindl** [comment]
> > [Warning] AMICI:NaN: AMICI encountered a NaN value at index 12/122 in p!
> 
> This looks fishy

Not necessarily a problem - that might be an output parameter introduced during PEtab import that is not used/set in the given simulation condition.

**Comment by FFroehlich** [comment]
> > > [Warning] AMICI:NaN: AMICI encountered a NaN value at index 12/122 in p!
> > 
> > 
> > This looks fishy
> 
> Not necessarily a problem - that might be an output parameter introduced during PEtab import that is not used/set in the given simulation condition and would not influence the (steady-)state.

True, but in the context of 
```
[Warning] AMICI:NaN: AMICI encountered a NaN value at index 10/29 in fxdot!
[Warning] AMICI:NaN: AMICI encountered a NaN value at index 12/122 in p!
[Warning] AMICI:NaN: AMICI encountered a NaN value at index 0/16 in w!
[Warning] AMICI:CVODES:CVode:OTHER: AMICI ERROR: in module CVODES in function CVode : The right-hand side routine failed at the first call.
```
I would argue that this probably isn't the case here.

**Comment by dweindl** [comment]
Okay, I overlooked the `fxdot` NaN.

**Comment by FFroehlich** [comment]
> > > > [Warning] AMICI:NaN: AMICI encountered a NaN value at index 12/122 in p!
> > > 
> > > 
> > > This looks fishy
> > 
> > 
> > Not necessarily a problem - that might be an output parameter introduced during PEtab import that is not used/set in the given simulation condition and would not influence the (steady-)state.
> 
> True, but in the context of
> 
> ```
> [Warning] AMICI:NaN: AMICI encountered a NaN value at index 10/29 in fxdot!
> [Warning] AMICI:NaN: AMICI encountered a NaN value at index 12/122 in p!
> [Warning] AMICI:NaN: AMICI encountered a NaN value at index 0/16 in w!
> [Warning] AMICI:CVODES:CVode:OTHER: AMICI ERROR: in module CVODES in function CVode : The right-hand side routine failed at the first call.
> ```
> 
> I would argue that this probably isn't the case here.

Oh, looks like parameter table is only 11 entries long, so the two are probably not connected.

**Comment by FFroehlich** [comment]
@jvanhoefer I think the issue is probably w[0] then, try to figure out how that value could end up being None by looking at the generated w.cpp.

Also pull the latest commit from develop since I just merged a critical fix for nested assignment rules (although I would expect a different error there, but you never know).

**Comment by dweindl** [comment]
Problem is, that your initial state has NaNs. With the NaNs in the condition table, you want to say that the initial concentration from the SBML model should be used, and that they should not be reinitialized after preequilibration, right?
The NaNs from the condition table might not be handled correctly. Seems like we need some additional test cases for that.

**Comment by jvanhoefer** [comment]
Yes, that sounds plausible. 

I wanted the model to take the preequilibration as initial state, in case I specified the initial values to be NaN. Otherwise the numeric value should be taken.

**Comment by jvanhoefer** [comment]
Which also would explain, why just simulating the model with the initial values given in the SBML works...

**Comment by dweindl** [comment]
Should be fixed in 0.11.13
